basicdataset/src/main/java/ai/djl/basicdataset/nlp/StanfordQuestionAnsweringDataset.java

```diff
@@ -241,13 +241,19 @@ public Object getData() throws IOException {
     @Override
     protected void preprocess(List<String> newTextData, boolean source) throws EmbeddingException {
         TextData textData = source ? sourceTextData : targetTextData;
-        QuestionInfo questionInfo = questionInfoList.get(Math.toIntExact(this.limit) - 1);
-        int lastIndex =
-                source
-                        ? questionInfo.questionIndex
-                        : questionInfo.answerIndexList.get(questionInfo.answerIndexList.size() - 1);
-        textData.preprocess(
-                manager, newTextData.subList(0, Math.min(lastIndex + 1, newTextData.size())));
+        int limit;
```

Comment:
A few comments:
1. avoid use `limit` as local variable, it's confusing because it's clash with `this.limit`
2. to avoid ArrayOutOfBoundException, the following should just work:

```
        int index = Math.min(Math.toIntExact(limit), questionInfoList.size()) -1;
        QuestionInfo questionInfo = questionInfoList.get(index);
```

Reply:
OK, I'll modify this.